client/state/data-layer/utils.js

```diff
@@ -23,7 +23,25 @@ export const bypassDataLayer = action => extendAction( action, doBypassDataLayer
 /**
  * Deeply converts keys from the specified object to camelCase notation.
  *
- * @param {Object} - object to convert
- * @returns a new object with all keys converted
+ * @param {Object} obj object to convert
+ * @returns {Object}   a new object with all keys converted
  */
-export { humps as convertToCamelCase };
+export function convertToCamelCase( obj ) {
+	if ( isArray( obj ) ) {
+		return map( obj, convertToCamelCase );
+	}
+
+	if ( isPlainObject( obj ) ) {
+		return reduce(
+			obj,
+			( result, value, key ) => {
+				const newKey = camelCase( key );
+				const newValue = isObjectLike( value ) ? convertToCamelCase( value ) : value;
+				return set( result, newKey, newValue );
```

Comment:
why `set` over `result[ newKey ] = newValue`?

Reply:
It was used in the original [lodash-humps code](https://github.com/cape-io/lodash-humps/blob/master/src/createHumps.js#L5) I'm replacing, so I left it as it is.

And I was really pleased to learn that `_.set` returns the modified object, making the common reduce-into-object pattern very elegant to implement:
```js
reduce( [ 1, 2, 3 ], ( accu, x ) => set( accu, `prop-${x}`, x ) );
```

Reply:
yeah, just minorly concerned that `set` does magic with keys with paths or indexes in them. We don't expect that here since `camelCase` should remove any such things...

Reply:
There is a unit test that checks that keys like `with.dot` or `with[bracket]` are correctly converted: `camelCase` removes all the special chars.

But I modified the `set` call to `set( result, [ newKey ], newValue )`, i.e., put the key into brackets. That makes it perfectly safe.

Reply:
oh neat trick!